### DISCUSSION TITLE
访问速度很慢

### DISCUSSION THREAD

**Discussion by 1105570390** [halo-dev/halo#6999] [Q&A]
### 系统信息

- 外部访问地址: http://47.107.***.***:8090
- 启动时间: 2024-10-18 13:15
- 版本: 2.19.3
- 构建时间: 2024-09-10 17:45
- Git Commit: 9ffb1bb
- Java: IBM Semeru Runtime Open Edition / 21.0.4+7-LTS
- 数据库: MySQL Community Server - GPL / 8.0.16
- 操作系统: Linux / 5.10.134-15.al8.x86_64
- 已激活主题: [Earth 1.9.0](https://github.com/halo-dev/theme-earth)
- 已启动插件:
  - [静态网页服务 1.0.0-alpha.5](https://www.halo.run/store/apps/app-gFkMn)
  - [页面静态缓存 1.2.1](https://github.com/halo-sigs/plugin-page-cache)
  - [PostChat 1.1.0](https://github.com/zhheo/halo-plugin-postchat)
  - [AI 助手 1.0.0](https://www.halo.run/store/apps/app-riNgb)
  - [live2d 看板娘 1.2.1](https://github.com/LIlGG/plugin-live2d)
  - [通知我 1.1.0](https://github.com/monyuan/notify-me)
  - [文章导入导出 1.2.4](https://jhacker.cn/2023/halo-plugin-export2doc)
  - [站点推送插件 1.5.2](https://github.com/Stonewuu/halo-plugin-sitepush)
  - [编辑器超链接卡片 1.0.3](https://github.com/halo-sigs/plugin-editor-hyperlink-card)
  - [highlight.js 代码高亮 1.3.2](https://github.com/halo-sigs/plugin-highlightjs)
  - [Sitemap 1.1.2](https://github.com/halo-dev/plugin-sitemap)
  - [搜索组件 1.6.2](https://github.com/halo-dev/plugin-search-widget)
  - [RSS 1.3.0](https://github.com/halo-dev/plugin-feed)
  - [评论组件 2.4.0](https://github.com/halo-dev/plugin-comment-widget)
  - [应用市场 1.4.0](https://www.halo.run/store/apps/app-VYJbF)

### 使用的哪种方式运行？

Docker Compose

### 发生了什么？

访问网站很慢，需要六七秒才能打开，有时候更久，没做cdn，服务器是阿里云服务器，是2h2g3m的服务器，磁盘是40G(2120 IOPS)，主题是官方默认的主题，在未安装其他插件的情况下，也就是构建完docker容器运行起来服务之后，访问就很慢了
![Image](https://github.com/user-attachments/assets/ae7f31bd-f25d-4270-a456-fcbd27f64542)

### 复现步骤

1.直接打开http://47.107.***.***:8090
2.无论是打开还是刷新速度都很慢

**Comment by ruibaby** [comment]
请提供更多信息，比如部署配置，数据库配置等。

**Comment by 1105570390** [comment]
> 请提供更多信息，比如部署配置，数据库配置等。

部署配置：docker的服务器2h2g3m的服务器，磁盘是40G(2120 IOPS)
数据库配置： MySQL Community Server - GPL / 8.0.16，数据库为外部数据库，外部数据库的服务器为1h2g1m的服务器，磁盘是40G(2120 IOPS)

**Comment by ruibaby** [comment]
如果你的数据库和 Halo 不在同一个网络，且延迟较高，那么综合下来就是会很慢。建议检查一下延迟。

**Comment by 1105570390** [comment]
> 如果你的数据库和 Halo 不在同一个网络，且延迟较高，那么综合下来就是会很慢。建议检查一下延迟。

修复了一下mysql的远程访问，但是主页访问仍然很慢，但是控制台的速度变快了
halo服务器与数据库服务器延迟测试图：
![Image](https://github.com/user-attachments/assets/0e4a0ccc-26b2-4b56-920f-3d11aa1f70e4)


主页图：
![Image](https://github.com/user-attachments/assets/8abd4b7e-4f0b-40c9-9957-56e4d0193712)


控制台图：
![Image](https://github.com/user-attachments/assets/c34fe9c5-5dec-4d23-83c8-c6ee648a6311)

**Comment by ruibaby** [comment]
30 多 ms 的延迟单看不高，但是目前页面上的查询不是并行的且需要等待所有数据查询完成才开始渲染页面，延迟加上数据传输就是会导致页面渲染过慢，Console 看起来快只是因为接口是异步的，并且单个接口可能不会造成太多的数据查询。

这个问题暂时没有好的办法解决，要么尝试和远程数据库进行内网互联，要么将数据库放在和 Halo 同一台服务器。
